internal_transfer_with_agreed_amount/wizards/internal_transfer_multicurrency.py

```diff
@@ -0,0 +1,85 @@
+from odoo import _, fields, models
+from odoo.exceptions import ValidationError
+
+
+class InternalTransferMulticurrency(models.TransientModel):
+    _name = "internal.transfer.multicurrency"
+    _description = "Wizard to modify the agreed amount with the bank on an internal transfer."  # noqa
+
+    agreed_amount = fields.Monetary(help="Agreed amount in the currency of the destination bank.")
+    currency_id = fields.Many2one(
+        "res.currency",
+        required=True,
+        default=lambda self: self._get_currency(),
+    )
+
+    def _get_currency(self):
+        active_ids = self._context.get("active_ids")
+        active_model = self._context.get("active_model")
+        if not active_ids or active_model != "account.payment":
+            return False
+        payment = self.env["account.payment"].browse(active_ids)
+        return payment.destination_journal_id.currency_id or payment.company_currency_id
+
+    def _validate_currencies_in_internal_transfer(self, payment):
+        payment_currency = payment.currency_id
+        company_currency = payment.company_currency_id
+        send_currency = payment.journal_id.currency_id or company_currency
+        receive_currency = self.currency_id
+        currencies = payment_currency | company_currency | send_currency | receive_currency
+        if len(currencies) == 2:
+            return currencies - payment_currency
+        raise ValidationError(
+            _(
+                """
+                Agreed amount is available when there are two considered currencies.
+                Currently, the %s currencies are considered for this process.
```

Comment:
The following currencies are being considered:
- EUR
- USD
- MXN

Otherwise, all those will be inserted inside the paragraph

Reply:
I do not get this comment.

Reply:
@luisg123v 

the validation is to avoid having a Transfer where only one or three currencies are involve:

if only one currency there is no case for using this feature.

If three currencies are involved, that is not supported.
That is the case where for example:
- Company: MXN.
- Source: EUR.
- Destination: USD.

Only Local to Foreing or Foreign to Local is supported.

cc/ @rolandojduartem 

Best Regards.

Reply:
I just mean a matter of formatting. Current message would be:

"Currently, the MXN, USD, EUR currencies are considered for this process"

Reply:
The idea of this message is not taking into account the quantity of currencies. It could be one or three, maybe more. How could we do it as you mentioned?

Reply:
@luisg123v Got it.

cc/ @rolandojduartem

Reply:
Either way is ok to me.

Let us set one way and move on.

@luisg123v @rolandojduartem

Reply:
@rolandojduartem 

> The idea of this message is not taking into account the quantity of currencies. It could be one or three, maybe more. How could we do it as you mentioned?

It could be done using also a mapped, but joining a list, e.g.:


`"currencies:\n- %s" % "\n- ".join(currencies.mapped("name"))`

Reply:
I am ok with both ways, either non-listed (original way) or listed (suggested way) is ok for me.

Thus, a this is not a blocking issue for this let us merge.

Therefore resolving this issue.

Best Regards.

Reply:
@luisg123v 

I am ok with both ways, either non-listed (original way) or listed (suggested way) is ok for me.

Thus, a this is not a blocking issue for this let us merge.

Therefore resolving this issue.

Best Regards.

cc/ @rolandojduartem @hugho-ad @desdelinux